node/src/main/kotlin/net/corda/node/services/statemachine/SingleThreadedStateMachineManager.kt

```
@@ -847,15 +847,20 @@ class SingleThreadedStateMachineManager(
                 val flowLogic = flow.fiber.logic
                 if (flowLogic.isEnabledTimedFlow()) scheduleTimeout(id)
                 flow.fiber.scheduleEvent(Event.DoRemainingWork)
-                when (checkpoint.flowState) {
-                    is FlowState.Unstarted -> {
-                        flow.fiber.start()
-                    }
-                    is FlowState.Started -> {
-                        Fiber.unparkDeserialized(flow.fiber, scheduler)
-                    }
-                }
+                startOrResume(checkpoint, flow)
+            }
+        }
+    }
+
+    private fun startOrResume(checkpoint: Checkpoint, flow: Flow) {
+        when (checkpoint.flowState) {
+            is FlowState.Unstarted -> {
+                flow.fiber.start()
+            }
+            is FlowState.Started -> {
+                Fiber.unparkDeserialized(flow.fiber, scheduler)
             }
+            null -> { } //Cannot start a flow with a null flow state.
```

This should throw an exception (probably an `IllegalStateException`)